relay-cardinality/src/cache.rs

```diff
@@ -119,22 +132,26 @@ impl<'a> CacheUpdate<'a> {
 #[derive(Debug, Default)]
 struct Inner {
     cache: hashbrown::HashMap<QuotaScoping, ScopedCache>,
-    current_slot: u64,
 }
 
 /// Scope specific information of the cache.
 #[derive(Debug, Default)]
-struct ScopedCache(
+struct ScopedCache {
     // Uses hashbrown for a faster hasher `ahash`, benchmarks show about 10% speedup.
-    hashbrown::HashSet<u32>,
-);
+    hashes: hashbrown::HashSet<u32>,
+    current_slot: Slot,
+}
 
 impl ScopedCache {
-    fn check(&self, hash: u32, limit: u64) -> CacheOutcome {
-        if self.0.contains(&hash) {
+    fn check(&self, slot: Slot, hash: u32, limit: u64) -> CacheOutcome {
+        if slot != self.current_slot {
+            return CacheOutcome::Unknown;
```

Comment:
We shouldn't hit this branch in prod, but should we emit e.g. an error in case we missed something?

Reply:
We hit this branch when you try to read from the cache with a new (or old) slot.

This should always be `Some` since we hold a `RwLockGuard` and it was previously inserted if missing, but I decided to write tests instead.
```rs
            if let Some(cache) = inner.cache.get_mut(key) {
                cache.insert(hash);
            }
```

Ideally I would get rid of this entirely which we can if we wrap the map value with `Arc<RwLock<>>` which I plan to revisit in the future.